src/viem/saga.ts

```diff
@@ -388,6 +384,9 @@ export function* sendAndMonitorTransaction({
         status: receipt.status === 'success',
       })
     )
+    if (receipt.status === 'reverted') {
+      throw new Error('transaction reverted')
+    }
```

Comment:
This change here affects the `transaction_confirmed` analytics event, just double checking, is this all ok?

Reply:
Oh, that's true, I'm gonna move the analytics after this to keep with the current logic.